### PR TITLE
Feature: #21669 - Guests Ignore Price Limits

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
Adds a cheat that allows for guests to ignore the prices of rollercoasters, shop items, etc.

### CODE REVIEW

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -3694,6 +3694,8 @@ STR_6621    :Restrict
 STR_6622    :Restrict object to the Scenario Editor and Sandbox mode.
 STR_6623    :Type ‘help’ for a list of available commands. Type ‘hide’ to hide the console.
 STR_6624    :Tile Inspector: Sort elements
+STR_6625    :Guests ignore prices
```
Comment (Harry-Hopkinson) [inline data/language/en-GB.txt:3697]: Was not too sure on the strings - "Guests ignore prices" may be too vague...

**Comment by Harry-Hopkinson** [comment]
I have noticed that some of the widgets do not appear to lineup properly so I will fix that soon.

File: src/openrct2/Cheats.h [hunk 2]
```diff
@@ -63,6 +64,7 @@ enum class CheatType : int32_t
     UnlockAllPrices,
     BuildInPauseMode,
     IgnoreRideIntensity,
+    IgnorePrice,
```
Comment (duncanspumpkin) [inline src/openrct2/Cheats.h:67]: This order should not be modified. You can only add to the end

File: src/openrct2/Cheats.cpp [hunk 3]
```diff
@@ -98,6 +99,7 @@ void CheatsSerialise(DataSerialiser& ds)
         CheatEntrySerialise(ds, CheatType::DisableAllBreakdowns, gameState.Cheats.DisableAllBreakdowns, count);
         CheatEntrySerialise(ds, CheatType::BuildInPauseMode, gameState.Cheats.BuildInPauseMode, count);
         CheatEntrySerialise(ds, CheatType::IgnoreRideIntensity, gameState.Cheats.IgnoreRideIntensity, count);
+        CheatEntrySerialise(ds, CheatType::IgnorePrice, gameState.Cheats.IgnorePrice, count);
```
Comment (duncanspumpkin) [inline src/openrct2/Cheats.cpp:102]: This order is important can only add to the end

**Comment by xrgtmx** [comment]
Would it be better to caption the cheat "Guests ignore **high** prices" ?

**Comment by ZehMatt** [comment]
> Would it be better to caption the cheat "Guests ignore **high** prices" ?

I think its fine the way it is, it makes them essentially ignore prices in general as there is no low price threshold in the game.

**Comment by Harry-Hopkinson** [comment]
Note that I have bumped the network/park version and added a changelog entry.

**Comment by spacek531** [comment]
I tested for about 30 minutes, toggling it on and off with prices set to $20.00. Guests refused to buy with the cheat off and would buy with the cheat on. I tested shop stalls, bathrooms, and rides. Enabling and disabling worked as expected.

**Review by spacek531 — APPROVED** [review]

**Comment by Harry-Hopkinson** [comment]
Thanks a lot for testing 😃 - I have also tested a lot on quite a few parks with shops, bathrooms and rides and have not noticed anything wrong with it.

**Comment by Harry-Hopkinson** [comment]
Yes, I have left the code untouched which checks if the RidePrice is greater than the total cash on the Peep,

**Comment by spacek531** [comment]
During my testing I checked on a few peeps and they didn't have negative money. I observed peeps thinking, "I don't have enough money for [item]." By inspection, the changes wouldn't break that because the check for guest money occurs earlier in the function in `DecideAndBuyItem`, `ShouldGoOnRide`, and later in the function for `ShouldGoToShop`

**Comment by Harry-Hopkinson** [comment]
Is this ready or is there anything else I need to do?

File: src/openrct2/localisation/StringIds.h [hunk 4]
```diff
@@ -3855,6 +3855,9 @@ enum : uint16_t
     STR_OPTIONS_TOOLBAR_BUTTONS_CENTRED = 6629,
     STR_OPTIONS_TOOLBAR_BUTTONS_CENTRED_TIP = 6630,
 
+    STR_CHEAT_IGNORE_PRICE = 6631,
+    STR_CHEAT_IGNORE_PRICE_TIP = 6632,
+
```
Comment (AaronVanGeffen) [inline src/openrct2/localisation/StringIds.h:3860]: As these are UI exclusive strings, these should now go into the new UiStringIds.h file in the openrct2-ui subproject.

Comment (Harry-Hopkinson) [reply src/openrct2/localisation/StringIds.h]: I have moved the strings over.

**Comment by Harry-Hopkinson** [comment]
Note that I have rebased this PR and moved the tip string to the UiStringIds.h file.

**Comment by Harry-Hopkinson** [comment]
Please keep open.

**Comment by Gymnasiast** [comment]
Could you rebase this?

**Comment by Harry-Hopkinson** [comment]
I have rebased this.

**Comment by Gymnasiast** [comment]
It looks like you’ll have to do it again. Sorry about that, we have been merging PRs like crazy today :')

**Comment by Harry-Hopkinson** [comment]
No worries, I have rebased it again.

File: src/openrct2/park/ParkFile.h [hunk 5]
```diff
@@ -14,7 +14,7 @@ namespace OpenRCT2
     constexpr uint32_t PARK_FILE_CURRENT_VERSION = 35;
 
     // The minimum version that is forwards compatible with the current version.
-    constexpr uint32_t PARK_FILE_MIN_VERSION = 33;
+    constexpr uint32_t PARK_FILE_MIN_VERSION = 34;
```
Comment (Gymnasiast) [inline src/openrct2/park/ParkFile.h:17]: This is not correct. You need to increase `PARK_FILE_CURRENT_VERSION` and leave this one as-is.

**Comment by AaronVanGeffen** [comment]
> No worries, I have rebased it again.

Sorry to say, #22490 has caused this to conflict again. It will have to be rebased once more, unfortunately.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**